packages/node_modules/@node-red/editor-client/src/js/ui/palette-editor.js

```diff
@@ -595,15 +597,10 @@ RED.palette.editor = (function() {
         }
     }
     function sortModulesRelevance(A,B) {
-        var currentFilter = searchInput.searchBox('value').trim();
-        if (currentFilter === "") {
-            return sortModulesAZ(A,B);
-        }
-        var i = A.info.index.indexOf(currentFilter) - B.info.index.indexOf(currentFilter);
-        if (i === 0) {
-            return sortModulesAZ(A,B);
-        }
-        return i;
+        return sortModulesDownloads(A,B);
```

Comment:
Something like?

```suggestion
        if (A.info.downloads) {
            return sortModulesDownloads(A,B);
        }

        var currentFilter = searchInput.searchBox('value').trim();
        if (currentFilter === "") {
            return sortModulesAZ(A,B);
        }
        var i = A.info.index.indexOf(currentFilter) - B.info.index.indexOf(currentFilter);
        if (i === 0) {
            return sortModulesAZ(A,B);
        }
        return i;
```

Reply:
The issue with resorting to this is the the sort is compare different elements on different fields, which would result in confusing result. We need to settle on a default count for elements without one, either `0`, they get filtered to the bottom of the search or `Number.MAX_SAFE_INTEGER`, whereby they're prioritised to the top.

I'm thinking we do the latter, as if a user has included a custom catalogue (without this information) then it's more likely they'd be looking for a result there.

Then, we need a secondary sort - which should be the default sort already in place today.